src/game/game.rs

```diff
@@ -151,22 +151,31 @@ impl Actor for Game {
 }
 
 #[derive(Serialize, Clone)]
-pub struct GameDataMessage{
-    field: String
+pub struct GamePlayersMessage{}
+#[derive(Serialize, Clone)]
+pub struct GameDataMessage{}
+
+impl actix::Message for GamePlayersMessage {
+    type Result = Arc<Vec<PlayerData>>;
 }
 
 impl actix::Message for GameDataMessage {
+    type Result = Arc<Mutex<GameData>>;
+}
+
+impl Handler<GamePlayersMessage> for Game {
     type Result = Arc<Vec<PlayerData>>;
+
+    fn handle(&mut self, _msg: GamePlayersMessage, _ctx: &mut Self::Context) -> Self::Result {
+        Arc::new(self.players.iter().map(|(_, p)| p.data.clone()).collect::<Vec<PlayerData>>())
+    }
 }
 
 impl Handler<GameDataMessage> for Game {
-    type Result = Arc<Vec<PlayerData>>;
+    type Result = Arc<Mutex<GameData>>;
```

Comment:
You are cloning the GameData, so wrapping it into a Mutex is not useful. Is it a trait problem which forced you to add a Mutex, or do you want the return type to be some kind of "shared data" ?

If you want a shared mutable data, this is NOT how you would do it.